src/cpp/server/server_context.cc

```diff
@@ -182,7 +185,14 @@ void ServerContext::TryCancel() const {
 }
 
 bool ServerContext::IsCancelled() const {
```

Comment:
ok thanks. 

I think it would help if you add this as a comment for `CheckCancelledAsync()` function - that we aren't plucking the tag here because we expect it to be delivered normally to the server context's completion queue (as a result of `AsyncNotifyWhenDone()` being called).. 

Btw, overall this PR has been very educational to me. thanks! :)